src/main/java/picard/illumina/CheckIlluminaDirectory.java

```diff
@@ -113,30 +125,89 @@ protected int doWork() {
         log.info("Expected cycles: " + StringUtil.intValuesToString(expectedCycles));
 
         for (final Integer lane : LANES) {
-            IlluminaFileUtil fileUtil = new IlluminaFileUtil(BASECALLS_DIR, lane);
-            final List<Integer> expectedTiles = fileUtil.getExpectedTiles();
-            if (!TILE_NUMBERS.isEmpty()) {
-                expectedTiles.retainAll(TILE_NUMBERS);
-            }
+            if (USE_NEW_CONVERTER) {
+                List<Integer> tiles = new ArrayList<>();
+
+                File laneDir = new File(BASECALLS_DIR, IlluminaFileUtil.longLaneStr(lane));
+
+                File[] cycleDirs = IOUtil.getFilesMatchingRegexp(laneDir, IlluminaFileUtil.CYCLE_SUBDIRECTORY_PATTERN);
+
+                //check all bcls/cbcls
+                List<File> cbcls = new ArrayList<>();
+                Arrays.asList(cycleDirs)
+                        .forEach(cycleDir -> cbcls.addAll(
+                                Arrays.asList(IOUtil.getFilesMatchingRegexp(
+                                        cycleDir, "^" + IlluminaFileUtil.longLaneStr(lane) + "_(\\d{1,5}).cbcl$"))));
+                IOUtil.assertFilesAreReadable(cbcls);
+
+                //check all pf filter files
+                Pattern laneTileRegex = Pattern.compile(ParameterizedFileUtil.escapePeriods(
+                        ParameterizedFileUtil.makeLaneTileRegex(".filter", lane)));
+                File[] filterFiles = getTiledFiles(laneDir, laneTileRegex);
+                for (File filterFile : filterFiles) {
+                    Matcher tileMatcher = laneTileRegex.matcher(filterFile.getName());
+                    if (tileMatcher.matches()) {
+                        tiles.add(Integer.valueOf(tileMatcher.group(1)));
+                    }
+                }
+                IOUtil.assertFilesAreReadable(Arrays.asList(filterFiles));
+                tiles.sort(TILE_NUMBER_COMPARATOR);
+
+                //check s.locs
+                File locsFile = new File(BASECALLS_DIR.getParentFile(), "s.locs");
```

Comment:
Is there a benefit in doing that as a separate step? The file is being read from when the `LocsFileReader` constructor is called, so an IOException will be thrown.